### PR TITLE
fix(transport/grpc): add WithDiscoveryOptions for grpc client

### LINKED ISSUE #2363: grpc/resolver/discovery/resolver.go中日志关闭问题
#### What happened:
#1935 
#1942 
在上诉issue中,在discovery中新增了DisableDebugLog()对日志输入进行关闭,但discovery.NewBuilder()是在grpc client创建的内部中调用的
`	if options.discovery != nil {
		grpcOpts = append(grpcOpts,
			grpc.WithResolvers(
				discovery.NewBuilder(
					options.discovery,
					discovery.WithInsecure(insecure),
				)))
	}`
无法直接在外部对日志进行关闭, 提议在 grpc.DialInsecure()中增加关闭discovery日志的options
#### Environment:
- Kratos version (use `kratos -v`): v2.5.0

### PR DESCRIPTION
**Pull request by deagon**
fixes #2363

### CODE REVIEW

**Comment by haiyux** [comment]
please fix test

**Comment by deagon** [comment]
> please fix test

It's timeout. How to recheck?

**Comment by deagon** [comment]
@haiyux PTAL.

File: transport/grpc/client.go [hunk 1]
```diff
@@ -136,11 +144,14 @@ func dial(ctx context.Context, insecure bool, opts ...ClientOption) (*grpc.Clien
...
 	if options.discovery != nil {
+		if insecure {
+			options.discoveryOpts = append(options.discoveryOpts, discovery.WithInsecure(insecure))
+		}
 		grpcOpts = append(grpcOpts,
 			grpc.WithResolvers(
 				discovery.NewBuilder(
 					options.discovery,
-					discovery.WithInsecure(insecure),
+					options.discoveryOpts...,
```
Comment (shenqidebaozi) [inline transport/grpc/client.go:154]: missing default configuration

**Pull request closed without merging**